fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -636,19 +636,20 @@ where
             length,
         )?;
 
-        // Update frame pointer, if we have a stack frame (e.g. fp > 0)
-        if fp > 0 {
-            let size = CallFrame::code_size_offset().saturating_add(WORD_SIZE);
-
-            let old_code_size = Word::from_be_bytes(self.memory.read_bytes(fp)?);
-
+        // Update frame code size, if we have a stack frame (i.e. fp > 0)
+        if self.context.is_internal() {
+            let code_size_ptr =
+                (*self.fp).saturating_add(CallFrame::code_size_offset() as Word);
+            let old_code_size =
+                Word::from_be_bytes(self.memory.read_bytes(code_size_ptr)?);
+            let old_code_size = padded_len_word(old_code_size)
+                .expect("Code size cannot overflow with padding");
```

Comment:
I would assume the size should already be padded when written in memory. Is it really correct to just compute padding here?

Reply:
Our code uses unpadded value in the `CallFrame`

<img width="729" alt="image" src="https://github.com/FuelLabs/fuel-vm/assets/18346821/d01e21dc-5396-4740-8177-28bb26ea329f">

Reply:
Yep. Shouldn't `CallFrame` use the padded value?

Reply:
I don't know=) The question is to the compiler team: do they use it somehow or not? Or is it used by anyone=)

Our specification says "Code size in bytes.", that sounds like exact size of the contract without padding.

<img width="874" alt="image" src="https://github.com/FuelLabs/fuel-vm/assets/18346821/ee83a517-ec00-4f0d-91bc-d0ddf77debb3">

Reply:
I'll ask the team, but a quick search from Sway codebase seems like they don't use this for anything. Time to remove? :D

Reply:
Doesnt "word-aligned" mean the same thing as padding here?

Reply:
Yep.

Moved to use padded size in 7335f9fe.